app/views/base.py

```diff
@@ -37,6 +37,93 @@ class BaseView(django.views.View):
         'lang': utils.strip,
     }
 
+    class Env(object):
+        """Class to store environment information used by views and templates.
+        """
+        # pylint: disable=attribute-defined-outside-init
+        # pylint: disable=too-many-instance-attributes
+
+        @property
+        def action(self):
+            """Gets the action ID, an identifier for the page being served."""
+            return self._action
+
+        @action.setter
+        def action(self, value):
+            self._action = value
+
+        @property
+        def charset(self):
+            """Gets the character set being used to serve the page."""
+            return self._charset
+
+        @charset.setter
+        def charset(self, value):
+            self._charset = value
+
+        @property
+        def config(self):
+            """Gets the config, a config.Config object for the repository."""
+            return self._config
+
+        @config.setter
+        def config(self, value):
+            self._config = value
+
+        @property
+        def enable_javascript(self):
+            "Gets whether or not to enable JavaScript." ""
+            return self._enable_javascript
+
+        @enable_javascript.setter
+        def enable_javascript(self, value):
+            self._enable_javascript = value
+
+        @property
+        def global_url(self):
+            """Gets the URL for the global root."""
+            return self._global_url
+
+        @global_url.setter
+        def global_url(self, value):
+            self._global_url = value
+
+        @property
+        def lang(self):
+            """Gets the two-letter code for the language being used."""
```

Comment:
Since it is not always 2-letter (e.g., "prs", "zh-TW"), may be better to just call it "language code" and refer to const.py.